### PR TITLE
new functions: `packstack`, `anystack`, `allstack`, `dict_to_getter_with_default`, `prepare_and_apply`

### PR DESCRIPTION
**Pull request by uriva**

### CODE REVIEW

File: gamla/functional.py [hunk 1]
```diff
@@ -851,3 +850,19 @@ def partition_all(seq):
         return toolz.partition_all(n, seq)
 
     return partition_all
+
+
+@currying.curry
+def dict_to_getter_with_default(default, d: Dict):
```
Comment (0xnurl) [inline gamla/functional.py:856]: 👏

File: gamla/functional_generic.py [hunk 2]
```diff
@@ -699,3 +699,26 @@ def count_by(f: Callable) -> Dict[Any, int]:
...
+def prepare_and_apply(f):
+    """Transforms a higher order function to a regular one.
+
+    Uses the given value once to prepare a regular function, then to call it.
+
+    >>> def increment(x): return x + 1
+    >>> def decrement(x): return x - 1
+    >>> def conditional_transformation(x):
+            return increment if x < 10 else decrement
+    >>> prepare_and_transform(conditional_transformation)(15)
```
Comment (0xnurl) [inline gamla/functional_generic.py:721]: name mismatch

File: gamla/functional_generic.py [hunk 3]
```diff
@@ -699,3 +699,26 @@ def count_by(f: Callable) -> Dict[Any, int]:
...
+
+#: Like `stack` but doesn't require additional brackets.
+packstack = compose_left(functional.pack, stack)
+#: Runs `packstack` with given functions, then runs `all` on the output.
+allstack = compose_left(packstack, after(all))
+#: Runs `packstack` with given functions, then runs `any` on the output.
+anystack = compose_left(packstack, after(any))
+
+
+def prepare_and_apply(f):
```
Comment (0xnurl) [inline gamla/functional_generic.py:712]: not sure about "prepare" here. maybe something like: "squash_higher_order_func" ?

in any case this looks like black magic :)

**Review by 0xnurl — APPROVED** [review]

File: gamla/functional_generic.py (hunk 3 above)
Comment (uriva) [reply gamla/functional_generic.py:712]: a common exapmle i'm seeing now is look at a kg to understand what change you want to do, then actually do the change

Comment (uriva) [reply gamla/functional_generic.py:712]: so you run the function once on the input to prepare it, then again to actually transform the value

**Pull request merged**